tests/tape/test_qnode.py

```diff
@@ -330,6 +330,52 @@ def func(x):
             == "<QNode: wires=1, device='default.qubit.autograd', interface='autograd', diff_method='best'>"
         )
 
+    @pytest.mark.parametrize("par", [None, 1, 1.1, np.array(1.2)])
+    def test_diff_method_none(self, par):
+        """Test if diff_method=None works as intended."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev, diff_method=None)
+        assert qn.interface is None
+
+        grad = qml.grad(qn)
+
+        # Raise error in all cases
+        # Case 1: No input
+        # Case 2: int input
+        # Case 3: float input
+        # Case 4: numpy input
+        with pytest.raises(TypeError) as exp:
+            grad() if par is None else grad(par)
+
+    @pytest.mark.parametrize("par", [1, 1.1, np.array(1.2)])
+    def test_diff_method_none_no_qnode_param(self, par):
+        """Test if diff_method=None works as intended."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func():
+            qml.PauliX(wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev, diff_method=None)
+        assert qn.interface is None
+
+        grad = qml.grad(qn)
+
+        # No differentiation required. No error raised.
+        grad()
+
+        # Raise error
+        # Case 1: int input
+        # Case 2: float input
+        # Case 3: numpy input
+        with pytest.raises(TypeError) as exp:
+            grad(par)

```

Comment:
@glassnotes we can remove these if that's better. What do you suggest?

Reply:
Hi @ankit27kh , was actually just about to suggest this. The point was just that there is a mismatch in terms of what error is being caught, and here it is that `grad(par)` will throw an error simply because it has an unexpected argument.

I think the test case to include instead is a QNode that accepts parameter, but then is not used, so running `grad` does not throw an error.

Reply:
@glassnotes 
doing something like this:
```
def func(x):
    qml.PauliX(0)
    return qml.expvalue(qml.PauliZ(0))

.
.
grad()
```
will cause an error because it expects `x`.
Then we'll have to use `func(x=0)` in the definition. Now, `grad()` will pass. But, we are already passing a default value of `x=0`. It's going in circles.

Best approach to check that `grad` works without input will be just to remove lines 372-378

Reply:
You could do as written above, but then call it with an actual parameter: 

```
@qml.qnode(dev, diff_method=None)
def func(x):
    qml.PauliX(wires=0)
    return qml.expval(qml.PauliZ(0))

g = qml.grad(func)
g(0.3)
```

This should give something like `UserWarning: Output seems independent of input.`. Alternatively, removing lines 372-378 as you suggest also works. We really just want to test here that `grad` can be safely called in situations where there is no dependence on the parameter, and having no parameter (as is done in your test) is one way to do so :+1:

Reply:
I decided to go with the `grad()` approach. I have deleted the extra lines of code.